app/src/main/java/es/situm/gettingstarted/drawroutegeojson/DrawRouteGeojsonActivity.java

```diff
@@ -0,0 +1,250 @@
+import com.google.android.material.snackbar.Snackbar;
+import com.google.maps.android.data.geojson.GeoJsonLayer;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import es.situm.gettingstarted.R;
+import es.situm.gettingstarted.common.SampleActivity;
+import es.situm.sdk.SitumSdk;
+import es.situm.sdk.directions.DirectionsRequest;
+import es.situm.sdk.error.Error;
+import es.situm.sdk.location.util.CoordinateConverter;
+import es.situm.sdk.model.cartography.Building;
+import es.situm.sdk.model.cartography.Floor;
+import es.situm.sdk.model.cartography.Point;
+import es.situm.sdk.model.directions.Route;
+import es.situm.sdk.model.location.Angle;
+import es.situm.sdk.model.location.Bounds;
+import es.situm.sdk.model.location.CartesianCoordinate;
+import es.situm.sdk.model.location.Coordinate;
+import es.situm.sdk.utils.Handler;
+
+public class DrawRouteGeojsonActivity
+        extends SampleActivity
+        implements OnMapReadyCallback {
+
+    private final String TAG = getClass().getSimpleName();
+
+    private ProgressBar progressBar;
+    private FloatingActionButton fabDisplaySource;
+    private GoogleMap googleMap;
+    private Building building;
+    private Marker markerDestination;
+    private Marker markerOrigin;
+    private String floorId;
+    private Point pointOrigin;
+    private CoordinateConverter coordinateConverter;
+    private GeoJsonLayer routeLayer;
+    private JSONObject routeLayerSource;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_draw_route_geojson);
+        getSupportActionBar().setSubtitle(R.string.tv_select_points);
+        building = getBuildingFromIntent();
+        setup();
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+    }
+
+    @Override
+    protected void onDestroy() {
+        SitumSdk.navigationManager().removeUpdates();
+        super.onDestroy();
+    }
+
+    @Override
+    public void onMapReady(@NonNull final GoogleMap googleMap) {
+        this.googleMap = googleMap;
+        fetchFirstFloorImage(building, new Callback() {
+            @Override
+            public void onSuccess(Bitmap floorImage) {
+                coordinateConverter = new CoordinateConverter(building.getDimensions(), building.getCenter(), building.getRotation());
+                drawBuilding(floorImage);
+            }
+
+            @Override
+            public void onError(Error error) {
+                Snackbar.make(findViewById(R.id.container), error.getMessage(), Snackbar.LENGTH_INDEFINITE).show();
+            }
+        });
+
+        this.googleMap.setOnMapClickListener(latLng -> {
+            if (pointOrigin == null) {
```

Comment:
Probando a rama no móbil, conseguín ter 3 puntos á vez. Esto pasa porque salta o Toast "There is no route that satisfies the specified restrictions" pero non se controla e créase o punto de destino igualmente. Para solucionalo: 

1- Simplemente añadir unha condición tendo en conta esto xunto con este if
2- en clearMap() ter en conta esto e borrar todos os markers, en lugar de só o de origen e destino

![Screenshot from 2022-06-15 11-55-13](https://user-images.githubusercontent.com/103040497/173800647-75edcd7f-1e2e-426f-ba1b-94de6417649a.png)